content/en/docs/concepts/services-networking/network-policies.md

```diff
@@ -312,6 +323,78 @@ The value of the label is the namespace name.
 While NetworkPolicy cannot target a namespace by its name with some object field, you can use the
 standardized label to target a specific namespace.
 
+## Pod lifecycle
+
+When a new NetworkPolicy object is created, it may take some for a network plugin
+to handle the new object. If the pod that is affected by a NetworkPolicy
+is created before the network plugin has completed NetworkPolicy handling,
+that pod may be started unprotected, and isolation rules will be applied when 
+the NetworkPolicy handling is completed.
+
+Once the NetworkPolicy is handled by a network plugin,
+
+1. All newly created pods affected by a given NetworkPolicy will be isolated before 
+they are started.
+
+2. Allow rules will be applied eventually after the isolation rules (that includes applying both at the same time).
+In the worst case, a newly created pod may have no network connectivity at all, if
+isolation rules were already applied, but no allow rules were applied yet.
+
+Every created NetworkPolicy will be handled by a network plugin eventually, but there is no
+way to tell from the Kubernetes API when exactly that happens. 
```

Comment:
```suggestion
Every created NetworkPolicy will be handled by a network plugin eventually, but there is no
way to tell from the Kubernetes API when exactly the policy is being enforced for new or
existing Pods.
(As an example: if you write a NetworkPolicy implementation that begins enforcing the policy from
midnight the next day, that conforms to Kubernetes' requirements - but of course could be very
frustrating for cluster administrators).
```

Reply:
This suggestion looks more implementor-focused to me, and I think we were going to outline implementor guideline in a different document, wdyt?
Another note - I don't think the implementation you described will pass conformance tests for network policy, so maybe it is not really conformant?

Reply:
There are no _conformance_ tests for NetworkPolicy, but it wouldn't pass the ordinary e2e tests.

> I think we were going to outline implementor guideline in a different document

yeah, there's been discussion about that (partly inspired by this PR). https://groups.google.com/g/kubernetes-sig-network/c/Ga9SWGs00k4 / https://github.com/kubernetes/community/issues/7421